### DISCUSSION TITLE
Is rendered notification

### DISCUSSION THREAD

**Discussion by JanRaadts** [neuefische/questions#451] [Web]
hey guys,
I have the following question:
My app is based on a mapbox map, which sometimes takes a relatively long time to load. Is it possible to get a confirmation when the map / component is completely rendered? I would then show a loading screen up to that point.

**Comment by mpagels** [comment]
I found the [MapDataEvent](https://docs.mapbox.com/mapbox-gl-js/api/events/#mapdataevent) on the [https://docs.mapbox.com](https://docs.mapbox.com) site. (I'm assuming that is the library you are using??)

And the event looks like this:

```
// Example of a MapDataEvent of type "sourcedata"
map.on('sourcedata', (e) => {
console.log(e);
// {
//   dataType: "source",
//   isSourceLoaded: false
//   source: {
//     type: "vector",
//     url: "mapbox://mapbox.mapbox-streets-v8,mapbox.mapbox-terrain-v2"
//   },
//   sourceDataType: "visibility",
//   sourceId: "composite",
//   style: {...},
//   target: {...},
//   type: "sourcedata"
// }
});
```

Checking for this property `isSourceLoaded` could help to display a loading indicator or the loaded map.
